The same patch as here:
https://bugzilla.mozilla.org/show_bug.cgi?id=1444953

Fix build on -current after 1200059

--- deps/erlang_js/c_src/patches/nspr-src-ptsynch.c.patch	1970-01-01 03:00:00 UTC
+++ deps/erlang_js/c_src/patches/nspr-src-ptsynch.c.patch
@@ -0,0 +1,12 @@
+--- c_src/nsprpub/pr/src/pthreads/ptsynch.c.orig	2009-03-19 00:29:32.000000000 +0300
++++ c_src/nsprpub/pr/src/pthreads/ptsynch.c		2018-07-10 14:59:52.818165000 +0300
+@@ -828,7 +828,8 @@
+  * From the semctl(2) man page in glibc 2.0
+  */
+ #if (defined(__GNU_LIBRARY__) && !defined(_SEM_SEMUN_UNDEFINED)) \
+-    || defined(FREEBSD) || defined(OPENBSD) || defined(BSDI) \
++    || (defined(FREEBSD) && __FreeBSD_version < 1200059) \
++    || defined(OPENBSD) || defined(BSDI) \
+     || defined(DARWIN) || defined(SYMBIAN)
+ /* union semun is defined by including <sys/sem.h> */
+ #else
